darkmoon/api/v1/metadata/views.py

```diff
@@ -95,26 +96,15 @@ async def get_metadata_by_id(
         if doc:
             document = MetadataEntity(**doc)
         else:
-            raise HTTPException(status_code=404, detail="Item not found")
+            raise ItemNotFoundException(status_code=404, detail="Item not found.")
 
         return document
 
     except errors.ServerSelectionTimeoutError:
-        raise HTTPException(
-            status_code=408,
-            detail=(
-                "The computer can't find the server.",
-                "Check the IP Address and the server name.",
-            ),
-        )
+        raise ServerNotFoundException(status_code=504, detail="Server timed out.")
+
     except bson.errors.InvalidId:  # type: ignore
-        raise HTTPException(
-            status_code=400,
-            detail=(
-                "This is not a valid ID.",
-                "It must be a 12-byte input or a 24-character hex string.",
-            ),
-        )
+        raise InvalidIDException(status_code=400, detail="Invalid ID")
```

Comment:
This is likely also a 422, or if it's invalid as in unauthorized (which I don't think fits here) it would be a 401
[https://restfulapi.net/http-status-codes/](url)

Reply:
actually since you're asking for an object with a certain ID a 404 not found would also work here cause a random ID would not exist in the DB